app/helpers/application_helper/toolbar/basic.rb

```diff
@@ -44,9 +42,43 @@ def separator
     {:separator => true}
   end
 
+  def definition
+    unless @loaded
+      load_overrides
+      @loaded = true
+    end
+
+    @definition
+  end
+
   private
 
+  def extension_classes
+    return [] if self.class.name.nil?
+
+    #Vmdb::Plugins.instance.vmdb_plugins.collect do |plugin|
+    Vmdb::Plugins.collect do |plugin|
+      name = plugin.root.join(
+        'app/toolbar_overrides',
+        self.class.name.to_s.split('::').last.underscore
+      )
+      name = "#{name}.rb"
+      if File.exist?(name)
+        require name
+        instance = [plugin.name.sub('::Engine',''), 'ToolbarOverrides', self.class.name.to_s.split('::').last].join('::')
```

Comment:
> `plugin.name.sub('::Engine','')`

Slightly better: `plugin.name.chomp("::Engine")`

> self.class.name

What is an example of `self.class.name` here?  Trying to understand what this does....I think it might be easier with a simple .gsub of the middle section, but I'm not sure.